actors/common/src/main/java/org/sunbird/learner/actors/bulkupload/UserBulkMigrationActor.java

```diff
@@ -0,0 +1,259 @@
+        BulkMigrationUser migrationUser=prepareRecord(request,processId,migrationUserList);
+        ProjectLogger.log("UserBulkMigrationActor:processRecord:processing record for number of users ".concat(migrationUserList.size()+""));
+        insertRecord(migrationUser);
+    }
+
+    private void insertRecord(BulkMigrationUser bulkMigrationUser){
+        long insertStartTime=System.currentTimeMillis();
+        Map<String,Object>record=mapper.convertValue(bulkMigrationUser,Map.class);
+        long createdOn=System.currentTimeMillis();
+        record.put(JsonKey.CREATED_ON,new Timestamp(createdOn));
+        record.put(JsonKey.LAST_UPDATED_ON,new Timestamp(createdOn));
+        Response response=cassandraOperation.insertRecord(dbInfo.getKeySpace(), dbInfo.getTableName(), record);
+        ProjectLogger.log("UserBulkMigrationActor:insertRecord:time taken by cassandra to insert record of size ".concat(record.size()+"")+"is(ms):".concat((System.currentTimeMillis()-insertStartTime)+""));
+        sender().tell(response,self());
+    }
+    private BulkMigrationUser prepareRecord(Request request,String processID,List<MigrationUser>migrationUserList){
+        try {
+            String decryptedData=mapper.writeValueAsString(migrationUserList);
+            BulkMigrationUser migrationUser=new BulkMigrationUser.BulkMigrationUserBuilder(processID,decryptedData)
+                    .setObjectType(JsonKey.MIGRATION_USER_OBJECT)
+                    .setUploadedDate(ProjectUtil.getFormattedDate())
+                    .setStatus(ProjectUtil.BulkProcessStatus.NEW.getValue())
+                    .setRetryCount(RETRY_COUNT)
+                    .setTaskCount(migrationUserList.size())
+                    .setCreatedBy(getCreatedBy(request))
+                    .setUploadedBy(getCreatedBy(request))
+                    .build();
+                    return migrationUser;
+        }catch (Exception e){
+            e.printStackTrace();
+            ProjectLogger.log("UserBulkMigrationActor:prepareRecord:error occurred while getting preparing record with processId".concat(processID+""),LoggerEnum.ERROR.name());
+            throw new ProjectCommonException(
+                    ResponseCode.SERVER_ERROR.getErrorCode(),
+                    ResponseCode.SERVER_ERROR.getErrorMessage(),
+                    ResponseCode.SERVER_ERROR.getResponseCode());
+        }
+    }
+
+    private String getCreatedBy(Request request){
+        Map<String,String>data=(Map<String, String>) request.getRequest().get(JsonKey.DATA);
+        return MapUtils.isNotEmpty(data)?data.get(JsonKey.CREATED_BY):null;
+    }
+
+    private List<MigrationUser> validateRequestAndReturnMigrationUsers(String processId,byte[] fileData,Map<String,Object>fieldsMap){
+            Map<String, List<String>> columnsMap = (Map<String, List<String>>) fieldsMap.get(JsonKey.FILE_TYPE_CSV);
+            List<String>csvHeaders=getCsvHeadersAsList(fileData);
```

Comment:
This file requires a rewrite. I find the fileData bytes are being read more than once and may be few milliseconds would be saved if not done so. Also, I feel some of these code must be in another separate class.